File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.c
#include <string.h>

int maximumLengthSubstring(char* s) {
    int freq[26] = {0};

    int left = 0;
    int ans = 0;
    int n = strlen(s);

    for (int right = 0; right < n; right++) {

        int idx = s[right] - 'a';
        freq[idx]++;

        while (freq[idx] > 2) {
            freq[s[left] - 'a']--;
            left++;
        }

        int len = right - left + 1;

        if (len > ans)
            ans = len;
    }

    return ans;
}

File: 3090-maximum-length-substring-with-two-occurrences/3090-maximum-length-substring-with-two-occurrences.js
var maximumLengthSubstring = function(s) {
    const freq = new Array(26).fill(0);

    let left = 0;
    let ans = 0;

    for (let right = 0; right < s.length; right++) {
        const idx = s.charCodeAt(right) - 97;

        freq[idx]++;

        while (freq[idx] > 2) {
            freq[s.charCodeAt(left) - 97]--;
            left++;
        }

        ans = Math.max(ans, right - left + 1);
    }

    return ans;
};